### PR TITLE
Multiple Files Properties: Move Files And Folders Count next to the i…

### LINKED ISSUE #1225: Multiple Files Properties: Move Files And Folders Count next to the icon
Current:
Anotación 2020-06-24 105435

Proposal:
Anotación 2020-06-24 104830

---------------------------------------------
Anotación 2020-06-24 105603

### PR DESCRIPTION
**Pull request by oleitao**
Closes #1225

**Resolved / Related Issues**
Itemize resolved / related issues by this PR.
- Move files and folders Count next to the icon on Properties panel

Capture1

Capture2

### CODE REVIEW

**Review by winston-de — CHANGES_REQUESTED** [review]
Why are you using a TextBox for this? It makes much more sense to use a TextBlock for this.
image

**Comment by winston-de** [comment]
When a single folder is selected, it shows it's contents label instead of it's name. 
image

**Comment by winston-de** [comment]
Also, you may want to use a separate control for this then the item name TextBox, as you have now inadvertently removed the ability to rename items from the properties dialog.

**Comment by yaichenbaum** [comment]
The issue in question is no longer relevant since it was opened before we displayed the item name.

**Comment by oleitao** [comment]
@yair100 you closed and reopen that issue. I don't know if is to continue the development or that topic is no more required?

**Comment by yaichenbaum** [comment]
@oleitao Let's continue development on this.

**Comment by oleitao** [comment]
test1

**Comment by winston-de** [comment]
Looking good, just a few things.

Could you put the name text box back into it's previous position (centered). Like this:
image

Also, could you please center this label as well?
image

Lastly, the name text box seems to be missing from the shortcut tab. Could you please add it back?
image

**Comment by winston-de** [comment]
I'm getting a crash when opening the dialog, it looks like you are trying to use a TextBox style on a TextBlock.

```
2021-04-25 00:13:37.5082|ERROR|Files.App|Exception has been thrown by the target of an invocation.|System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> Windows.UI.Xaml.Markup.XamlParseException: The text associated with this error code could not be found.

Cannot apply a Style with TargetType 'Windows.UI.Xaml.Controls.TextBox' to an object of type 'Windows.UI.Xaml.Controls.TextBlock'. [Line: 91 Position: 17]
   at Windows.UI.Xaml.Application.LoadComponent(Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
   at Files.Views.PropertiesGeneral.InitializeComponent()
   at Files.Views.PropertiesGeneral..ctor()
   --- End of inner exception stack trace ---
   at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor)
   at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean skipCheckThis, Boolean fillCache)
   at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
   at Windows.UI.Xaml.Controls.Frame.Navigate(Type sourcePageType, Object parameter, NavigationTransitionInfo infoOverride)
   at Files.Views.Properties.NavigationView_SelectionChanged(NavigationView sender, NavigationViewSelectionChangedEventArgs args)
   at Windows.UI.Xaml.Controls.NavigationView.put_SelectedItem(Object value)
   at Files.Views.Properties.XamlBindingSetters.Set_Windows_UI_Xaml_Controls_NavigationView_SelectedItem(NavigationView obj, Object value, String targetNullValue)
   at Files.Views.Properties.Properties_obj1_Bindings.Update_TabGeneral(NavigationViewItem obj, Int32 phase)
   at Files.Views.Properties.Properties_obj1_Bindings.Update_(Properties obj, Int32 phase)
   at Files.Views.Properties.Properties_obj1_Bindings.Update()
   at Files.Views.Properties.Properties_obj1_Bindings.Initialize()
   at Files.Views.Properties.Properties_obj1_Bindings.Loading(FrameworkElement src, Object data)
```

**Comment by oleitao** [comment]
> I'm getting a crash when opening the dialog, it looks like you are trying to use a TextBox style on a TextBlock.
> 
> ```
> 2021-04-25 00:13:37.5082|ERROR|Files.App|Exception has been thrown by the target of an invocation.|System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> Windows.UI.Xaml.Markup.XamlParseException: The text associated with this error code could not be found.
> 
> Cannot apply a Style with TargetType 'Windows.UI.Xaml.Controls.TextBox' to an object of type 'Windows.UI.Xaml.Controls.TextBlock'. [Line: 91 Position: 17]
>    at Windows.UI.Xaml.Application.LoadComponent(Object component, Uri resourceLocator, ComponentResourceLocation componentResourceLocation)
>    at Files.Views.PropertiesGeneral.InitializeComponent()
>    at Files.Views.PropertiesGeneral..ctor()
>    --- End of inner exception stack trace ---
>    at System.RuntimeTypeHandle.CreateInstance(RuntimeType type, Boolean publicOnly, Boolean wrapExceptions, Boolean& canBeCached, RuntimeMethodHandleInternal& ctor)
>    at System.RuntimeType.CreateInstanceSlow(Boolean publicOnly, Boolean wrapExceptions, Boolean skipCheckThis, Boolean fillCache)
>    at System.Activator.CreateInstance(Type type, Boolean nonPublic, Boolean wrapExceptions)
>    at Windows.UI.Xaml.Controls.Frame.Navigate(Type sourcePageType, Object parameter, NavigationTransitionInfo infoOverride)
>    at Files.Views.Properties.NavigationView_SelectionChanged(NavigationView sender, NavigationViewSelectionChangedEventArgs args)
>    at Windows.UI.Xaml.Controls.NavigationView.put_SelectedItem(Object value)
>    at Files.Views.Properties.XamlBindingSetters.Set_Windows_UI_Xaml_Controls_NavigationView_SelectedItem(NavigationView obj, Object value, String targetNullValue)
>    at Files.Views.Properties.Properties_obj1_Bindings.Update_TabGeneral(NavigationViewItem obj, Int32 phase)
>    at Files.Views.Properties.Properties_obj1_Bindings.Update_(Properties obj, Int32 phase)
>    at Files.Views.Properties.Properties_obj1_Bindings.Update()
>    at Files.Views.Properties.Properties_obj1_Bindings.Initialize()
>    at Files.Views.Properties.Properties_obj1_Bindings.Loading(FrameworkElement src, Object data)
> ```

@winston-de  I already tried to replicate that error, but everything works fine. Can you describe what you did to get that error?

**Comment by winston-de** [comment]
@oleitao It may not always cause a crash, but there definitely are mismatched types in the styles.

File: Files/Views/Pages/PropertiesGeneral.xaml [hunk 1]
```diff
@@ -81,7 +81,17 @@
...
                 Visibility="{x:Bind ViewModel.ItemNameVisibility, Mode=OneWay}" />
-
+            
+            <TextBlock
+                x:Name="itemItemFileIconLabel"
+                x:Uid="PropertiesItemFileIconLabel"
+                Grid.Row="0"
+                Grid.Column="1"
+                VerticalAlignment="Center"
+                Style="{StaticResource PropertyValueTextBox}"
```
Comment (winston-de) [inline Files/Views/Pages/PropertiesGeneral.xaml:91]:
```suggestion
                Style="{StaticResource PropertyValueTextBlock}"
```

```PropertyValueTextBox``` is a ```TextBox``` style, please switch to ```PropertyValueTextBlock```

**Review by winston-de — CHANGES_REQUESTED** [review]

**Review by winston-de — CHANGES_REQUESTED** [review]
Can you please fix the following issues and resolve the merge conflicts?
- Count is shown when viewing properties of single folder
image

- Item name is missing for shortcuts
image

**Comment by yaichenbaum** [comment]
Closing this PR due to inactivity.

**Pull request closed without merging**